File: udemy/machine learning A-Z/Part 5 - Association Rule Learning/Section 28 - Apriori/apriori.R
# Apriori

# Data Preprocessing
# dataset = read.csv('Market_Basket_Optimisation.csv', header = FALSE)
# Use sparse matrix instead
# install.packages('arules')
library(arules)  # Load sparse matrix module
# Dataset contains transaction in a week
dataset = read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset, topN = 10)

# Training Apriori on the dataset
# Minimum support is 4% or (4*7)/7501 or at least four transactions in a week
# Minimum confidence is 20%
rules = apriori(data = dataset, parameter = list(support = 0.004, confidence = 0.2))

# Visualising the results
inspect(sort(rules, by = 'lift')[1:10])

File: udemy/machine learning A-Z/Part 5 - Association Rule Learning/Section 28 - Apriori/apriori.py
# Apriori

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Data Preprocessing
# Dataset contains transaction in a week
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)
transactions = []
for i in range(0, 7501):  # Create sparse matrix
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])

# # Look the first second sparse matrix of dataset
# print(transactions[0])
# print(transactions[1])

# Training Apriori on the dataset
from apyori import apriori
# support=(3*7)/7501 is at least three transactions in a week
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

# Visualising the results
results = list(rules)
